valkyrie/lib/valkyrie/storage_adapter.rb

```diff
@@ -84,7 +82,7 @@ def disk_path
       private
 
         def tmp_file_name
-          id.to_s.split('://').last
+          id.to_s.tr(':/', '__')
```

Comment:
I was seeing errors about not being able to find file `$TMPDIR/localhost:8983/rest/dev/ab/cd/ef/gh/abcdefgh`, so I just decided to flatten commas and slashes so now the tempfiles look like `$TMPDIR_localhost_8983_rest_dev_ab_cd_ef_gh_abcdefgh`